client/llb/fileop_test.go

```diff
@@ -0,0 +1,632 @@
+package llb
+
+import (
+	"testing"
+	"time"
+
+	"github.com/moby/buildkit/solver/pb"
+	digest "github.com/opencontainers/go-digest"
+	"github.com/stretchr/testify/require"
+)
+
+func TestFileMkdir(t *testing.T) {
+	t.Parallel()
+
+	st := Image("foo").File(Mkdir("/foo", 0700))
```

Comment:
`.File` doesn't seem like the right naming given it can actually do so much more. Having said that I'm not sure what would be better, `FileOp` perhaps, or even `FsOp` maybe?

Reply:
Looking at a subsequent usage:
```go
st := Image("foo").Dir("/etc").File(Mkdir("/foo", 0700).Mkdir("bar", 0600, WithParents(true)).Mkdir("bar/baz", 0701, WithParents(false)))
```
I'm left wondering if we coudn't arrange for this to be:
```
st := Image("foo").Dir("/etc").Mkdir("/foo", 0700).Mkdir("bar", 0600, WithParents(true)).Mkdir("bar/baz", 0701, WithParents(false)).Root()
```
i.e. dropping the `.File()` and instead chain at a higher level. That feels more in keeping with the existing llb client library stuff.

I suppose the desire for `.File()` is to chain multiple file ops into a single llb op?

Reply:
With doing actions directly with the state we can't predict what actions run as part of a single op and what runs separately. As the caching is based on op, this isn't quite meaningless. We could add aliases of course `state.Mkdir() -> state.File(Mkdir())` but I think it might be confusing and promote inefficient builds.